compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -769,6 +766,42 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             };
             let captures = format!("captures data from {}", arg);
 
+            if fn_returns.is_empty() {
+                // in case the return type of the method is a type alias
+                let _ : Option<None> = self
```

Comment:
Can you flatten these with early returns instead of combinators?

Reply:
Thanks for the suggestion. Modified.